vendor/brave-ios/Ledger/BATBraveLedger.mm

```diff
@@ -1472,6 +1550,11 @@ - (void)removePendingContribution:(const std::string &)publisher_key viewingId:(
                                                   completion:^(BOOL success) {
                                                     callback(success ? ledger::Result::LEDGER_OK :
                                                              ledger::Result::LEDGER_ERROR);
+                                                    for (BATBraveLedgerObserver *observer in self.observers) {
+                                                      if (observer.pendingContributionRemoved) {
+                                                        observer.pendingContributionRemoved(publisherID);
```

Comment:
Have you considered returning an array of contributions instead of using for loop and single value returns?

I wonder if there's any benefit to returning an array of values here and in other observer methods.

Reply:
Im not sure what you mean? I actually missed calling this method on `removeAllPendingContributions` which I just realized because I either need a new observer method `pendingContributionsCleared` or something along that lines, or I'd have to change this (`pendingContributionRemoved`) to accept an array of publisher IDs (if this is what you meant?) and query all thee contributions before deleting them all

Reply:
Yes I meant changing `pendingContributionRemoved` to accept an array.

So if let's say 100k pending contributions is deleted, we can handle them all at once(using batch delete and updating a tableView once instead doing the same for each record separately)

Reply:
Yeah I get it, in this case since we have a `removeAllPendingContributions` I will change this to accept an array of publisher ID's. When a single one is deleted it will just be an array of 1

Reply:
works for me, or you could audit which of these callbacks could be a performance issue and sending an array of values would be better, maybe it's not worth the effort